web/src/reducers/activities.js

```diff
@@ -367,137 +368,11 @@ const reducer = (state = initialState, action) => {
-            years: c.years.reduce(
-              (years, y) => ({
-                ...years,
-                [y.year]: +y.cost
-              }),
-              {}
-            )
-          })),
-          expenses: a.expenses.map(e => ({
-            key: generateKey(),
-            id: e.id,
-            category: e.category || '',
-            desc: e.description || '',
-            years: e.entries.reduce(
-              (years, y) => ({
-                ...years,
-                [y.year]: +y.amount
-              }),
-              {}
-            )
-          })),
-
-          standardsAndConditions: {
-            bizResults: a.standardsAndConditions.businessResults || '',
-            documentation: a.standardsAndConditions.documentation || '',
-            industry: a.standardsAndConditions.industryStandards || '',
-            interoperability: a.standardsAndConditions.interoperability || '',
-            keyPersonnel: a.standardsAndConditions.keyPersonnel || '',
-            leverage: a.standardsAndConditions.leverage || '',
-            modularity: a.standardsAndConditions.modularity || '',
-            minimizeCost: a.standardsAndConditions.minimizeCost || '',
-            mita: a.standardsAndConditions.mita || '',
-            mitigation: a.standardsAndConditions.mitigationStrategy || '',
-            reporting: a.standardsAndConditions.reporting || ''
-          },
-          quarterlyFFP:
-            a.quarterlyFFP && a.quarterlyFFP.length
-              ? a.quarterlyFFP.reduce(
-                  (quarterlyAcc, ffy) => ({
-                    ...quarterlyAcc,
-                    [ffy.year]: {
-                      1: {
-                        combined: ffy.q1.combined * 100,
-                        contractors: ffy.q1.contractors * 100,
-                        state: ffy.q1.state * 100
-                      },
-                      2: {
-                        combined: ffy.q2.combined * 100,
-                        contractors: ffy.q2.contractors * 100,
-                        state: ffy.q2.state * 100
-                      },
-                      3: {
-                        combined: ffy.q3.combined * 100,
-                        contractors: ffy.q3.contractors * 100,
-                        state: ffy.q3.state * 100
-                      },
-                      4: {
-                        combined: ffy.q4.combined * 100,
-                        contractors: ffy.q4.contractors * 100,
-                        state: ffy.q4.state * 100
-                      }
-                    }
-                  }),
-                  {}
-                )
-              : { ...arrToObj(action.apd.years, quarterlyFFPEntry()) },
-          meta: {
-            expanded: a.name === 'Program Administration'
-          }
-        };
+        // const activity = fromAPI(a, action.apd.years, {
+        //   quarterlyFFP: { ...arrToObj(action.apd.years, quarterlyFFPEntry()) }
+        // });
```

Comment:
Don't have to deserialize activities in the reducer at all - they are deserialized along with the rest of the APD.  Now when we select an APD, we just have to copy the activities into the `byKey` map.